src/main/java/teammates/common/util/ActivityLogEntry.java

```diff
@@ -219,7 +271,12 @@ public ActivityLogEntry(AccountAttributes userAccount, boolean isMasquerade, Str
         }
         
         role = changeRoleToAutoIfAutomatedActions(servletName, role);
-        id = googleId + formatTimeForId(new Date(time));
+        boolean isUnregisteredStudent = (googleId.contentEquals("Unknown") || googleId.contentEquals("Unregistered")) && student != null;
+        if (isUnregisteredStudent) {
+            id = student.email + "%" + student.course + "%" + formatTimeForId(new Date(time));
+        } else {
+            id = googleId + "%" + formatTimeForId(new Date(time));
+        }
```

Comment:
Also, may be see if the two constructors can be refactored to bring out what's common and what's different?